dipy/denoise/enhancement_kernel.pyx

```diff
@@ -0,0 +1,420 @@
+    @cython.boundscheck(False)
+    @cython.nonecheck(False)
+    @cython.cdivision(True)
+    cdef void create_lookup_table(self, test_mode=False):
+        """ Compute the look-up table based on the parameters set
+        during class initialization
+        """
+        self.estimate_kernel_size()
+
+        cdef:
+            double [:, :] orientations = np.copy(self.orientations)
+            cnp.npy_intp OR1 = orientations.shape[0]
+            cnp.npy_intp OR2 = orientations.shape[0]
+            cnp.npy_intp N = self.kernelsize
+            cnp.npy_intp hn = (N-1)/2
+            cnp.npy_intp angv, angr, xp, yp, zp
+            double [:] x
+            double [:] y
+            cdef double [:, :, :, :, ::1] lookuptablelocal
+            double kmax = self.kernelmax
+            double l1norm
+            double kernelval  
+
+        # For testing, only compute one orientation of r
+        if test_mode:
+            OR2 = 1
+
+        lookuptablelocal = np.zeros((OR1, OR2, N, N, N))
+        x = np.zeros(3)
+        y = np.zeros(3) # constant at (0,0,0)
+
+        with nogil:
+
+            for angv in range(OR1):
+                for angr in range(OR2):
+                    for xp in range(-hn, hn + 1):
+                        for yp in range(-hn, hn + 1):
+                            for zp in range(-hn, hn + 1):
+
+                                x[0] = xp
+                                x[1] = yp
+                                x[2] = zp
+
+                                lookuptablelocal[angv,
+                                                 angr,
+                                                 xp + hn,
+                                                 yp + hn,
+                                                 zp + hn] = self.k2(x, y, orientations[angr,:], orientations[angv,:])
+
+        # save to class member
+        self.lookuptable = lookuptablelocal
+
+    @cython.wraparound(False)
+    @cython.boundscheck(False)
+    @cython.nonecheck(False)
+    @cython.cdivision(True)
+    cdef void estimate_kernel_size(self):
+        """ Estimates the dimensions the kernel should
+        have based on the kernel parameters.
+        """
+
+        cdef:
+            double [:] x
+            double [:] y
+            double [:] r
+            double [:] v
+            double i
+
+        x = np.array([0., 0., 0.])
+        y = np.array([0., 0., 0.])
+        r = np.array([0., 0., 1.])
+        v = np.array([0., 0., 1.])
+
+        # evaluate at origin
+        self.kernelmax = self.k2(x, y, r, v);
+
+        with nogil:
+
+            # determine a good kernel size
+            i = 0.0
+            while True:
+                i += 0.1
+                x[2] = i
+                kval = self.k2(x, y, r, v) / self.kernelmax
+                if(kval < 0.1):
+                    break;
+
+        N = ceil(i) * 2
+        if N % 2 == 0:
+            N -= 1
+
+        print("Dimensions of kernel: %dx%dx%d" % (N, N, N))
```

Comment:
Might be worth adding a `verbose` option that controls whether or not this always gets printed.